src/Library/Library.fs

```diff
@@ -1,5 +1,130 @@
-﻿namespace Library
-
-module Say =
-    let hello name =
-        printfn "Hello %s" name
+﻿namespace Library
+
+
+module factorial =
+    let rec factorial n =
+        if n < 0 then
+            failwith "The factorial of negative numbers is not calculated by this program\n"
+        elif n = 0 || n = 1 then
+            1
+        else
+            n * factorial (n - 1)
+
+
+module sorts =
+
+    let swap (arr: 't[]) i j =
+        let elem = arr[i]
+        arr[i] <- arr[j]
+        arr[j] <- elem
+
+
+    let bubble_sort (arr: 't[]) =
+
+        let mutable array_len = arr.Length
+        let mutable swapped = true
+
+        while swapped do
+            swapped <- false
+
+            for i in 0 .. array_len - 2 do
+                if compare arr[i] arr[i + 1] > 0 then
```

Comment:
Where 'compare' function is defined?

Reply:
It is a built into library function that takes 2 arguments a and b and returns 
      1 if a > b
      0 if a = b
     -1 if a < b
It can compare integers, floats, chars (by their Unicode value), strings (in alphabet order, comparing each letter) and any other data types on which a compare operation can be defined.